client/client.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/rpc"
+	"os"
+	. "rtclbedit/curp"
+	. "rtclbedit/shared"
+)
+
+func main() {
+	args := os.Args
+	if len(os.Args) != 3 {
+		fmt.Println("[usage]: " + args[0] + " <identifier> <configuration file>")
+		os.Exit(1)
+	}
+
+	name := args[1]
+	filename := args[2]
+	node_map := Parse(filename)
+	master_map, witness_map, _ := ParseByRole(node_map)
+	master := Map_begin(master_map)
+	master_client, _ := rpc.Dial("tcp", master.Ip+":"+master.Port)
+	var witness_clients []*rpc.Client
+	for _, v := range witness_map {
+		c, _ := rpc.Dial("tcp", v.Ip+":"+v.Port)
+		witness_clients = append(witness_clients, c)
+	}
```

Comment:
Super nitpicky, but we can probably multithread this/do it concurrently

Reply:
`rpc.Dial` call blocks and need mutex lock for append so multi-threading won't speed it up. Actual multi-thread happens some line further down.